test/src/testPubSub.cpp

```diff
@@ -0,0 +1,267 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * SPDX-FileType: SOURCE
+ * SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <spdlog/spdlog.h>
+#include <up-client-zenoh-cpp/client/upZenohClient.h>
+#include <up-cpp/transport/builder/UAttributesBuilder.h>
+// #include <up-cpp/uri/serializer/LongUriSerializer.h>
+#include <up-cpp/uri/serializer/MicroUriSerializer.h>
+#include <gtest/gtest.h>
+
+#include <condition_variable>
+#include <mutex>
+#include <deque>
+#include <vector>
+#include <tuple>
+#include <chrono>
+#include <random>
+#include <sys/socket.h>
+#include <poll.h>
+#include <unistd.h>
+#include <signal.h>
+
+using namespace uprotocol::utransport;
+using namespace uprotocol::v1;
+using namespace uprotocol::uri;
+using namespace uprotocol::uuid;
+using namespace uprotocol::client;
+
+// capture and queue onReceive() for verification by the sender
+
+template <bool INTERPROC> 
+class MessageCapture : public UListener {
+public:
+    typedef std::vector<std::uint8_t>    Data;
+private:
+    std::condition_variable cv;
+    std::mutex mtx;
+    std::deque<Data>    queue;
+    int sp[2];
+public:
+    MessageCapture()
+    {
+        if constexpr (INTERPROC) {
+            socketpair(AF_UNIX, SOCK_DGRAM, 0, sp);
+        }
+    }
+
+    ~MessageCapture()
+    {
+        if constexpr (INTERPROC) {
+            close(sp[0]);
+            close(sp[1]);
+        }
+    }
+
+    void operator()(UMessage& message)
+    {
+        using namespace std;
+
+        if constexpr (INTERPROC) {
+            send(sp[0], message.payload().data(), message.payload().size(), 0);
+        }
+        else {
+            Data data(message.payload().size());
+            auto ptr = message.payload().data();
+            copy(ptr, ptr+message.payload().size(), data.data());
+            { 
+                lock_guard<mutex> lock(mtx);
+                queue.emplace_back(data);
+                cv.notify_one();
+            }
+        }
+    }
+
+    UStatus onReceive(UMessage& m) const override
+    {
+        using namespace std;
+        // scrape the const-ness off this thing
+        const_cast<MessageCapture*>(this)->operator()(m);
+        UStatus status;
+        status.set_code(UCode::OK);
+        return status;
+    }
+
+    std::tuple<size_t, Data, std::chrono::time_point<std::chrono::steady_clock>> get(size_t milliseconds_max = 1000)
```

Comment:
Minor: use `std::chrono` and chrono literals for time values
```suggestion
    std::tuple<size_t, Data, std::chrono::steady_clock::time_point>> get(std::chrono::milliseconds max_wait = 1s)
```

Reply:
I will address this in next checkin, but I'm not thrilled with chrono literals for default argument values, since it seems to pollute namespace. Can we discuss this after next update?